### PR TITLE
[ESP32] Support for ESP32C2 and changes to not initialized BLE for already provisioned nodes

### LINKED ISSUE #22969: [Platform] [ESP32] Do not initialize the BLE if WiFi/Thread is already provisioned
### Reproduction steps / Feature

```bash
BLE is initialized every time even if device is commissioned and BLE is not being used in either subsequent boot or after device gets commissioned.

We can have a config option which let user select whether to deinit BLE after commissioning and on boot if not to initialize BLE if WiFi/Thread is already provisioned.

This would free some RAM.
```

### Platform

esp32

### Platform Version(s)

_No response_

### Type

Manually tested with SDK

### (Optional) If manually tested please explain why this is only manually tested

Tested with enabling and disabling `CONFIG_USE_BLE_ONLY_FOR_COMMISSIONING` option. When enabled validated that BLE does deinit after commissioning and BLE do not initialize on boot if device is already provisioned.

### Anything else?

_No response_

### LINKED ISSUE #22970: [Platform] Support ESP32-C2 devices
### Reproduction steps

```bash
ESP32C2 was introduced some time earlier and supported on [IDF v5.0-beta1](https://github.com/espressif/esp-idf/tree/v5.0-beta1). Currently v5.0-beta1 is used for ESP32H2(Thread device), so extending the support in SDK to support ESP32C2.
```

### Platform

esp32

### Platform Version(s)

_No response_

### Type

Manually tested with SDK

### (Optional) If manually tested please explain why this is only manually tested

Tested the lighting-app/esp32, commissioned using chip-tool and tested the basic cluster control.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by shubhamdp**
# Issues
- Fixes #22969
- Fixes #22970

# Description
- BLE is initialized every time even if device is commissioned and BLE is not being used in either subsequent boot or after device gets commissioned. Added a config option which let user select whether to deinit BLE after commissioning and on boot if not to initialize BLE if WiFi/Thread is already provisioned.

- ESP32C2 was introduced some time earlier and supported on [IDF v5.0-beta1](https://github.com/espressif/esp-idf/tree/v5.0-beta1). Currently v5.0-beta1 is used for ESP32H2(Thread device), so extending the support in SDK to support ESP32C2.

- Removed the BLE device name from the log as it is not being advertised in the BLE advertisement data

- Removing the `sdkconfig.old` present in `all-clusters-minimal-app/esp32` as this is autogenerated and not required.

### CODE REVIEW

File: examples/lighting-app/esp32/sdkconfig.defaults.esp32h2 [hunk 1]
```diff
@@ -20,7 +20,7 @@ CONFIG_BT_ENABLED=y
 CONFIG_BT_NIMBLE_ENABLED=y
 CONFIG_BT_NIMBLE_EXT_ADV=n
 CONFIG_BT_NIMBLE_HCI_EVT_BUF_SIZE=70
-CONFIG_DEINIT_BLE_ON_COMMISSIONING_COMPLETE=n
+CONFIG_USE_BLE_ONLY_FOR_COMMISSIONING=n
```
Comment (dhrishi) [inline examples/lighting-app/esp32/sdkconfig.defaults.esp32h2:23]: Why is it set to `n`? I think it should be removed from here

**Review by dhrishi — APPROVED** [review]
Minor comment. LGTM otherwise

**Comment by dhrishi** [comment]
@shubhamdp I think we also need an issue to be filed for a PR

File: examples/lighting-app/esp32/sdkconfig.defaults.esp32h2 (hunk 1 above)
Comment (shubhamdp) [reply examples/lighting-app/esp32/sdkconfig.defaults.esp32h2:23]: Not sure, but it was there before so did not change it, just renamed.

**Comment by shubhamdp** [comment]
@dhrishi, PR template read the following, so thought of not putting an issue.

> If you do not have an issue number, please have a good description of the problem and the fix

File: examples/lighting-app/esp32/sdkconfig.defaults.esp32h2 (hunk 1 above)
Comment (dhrishi) [reply examples/lighting-app/esp32/sdkconfig.defaults.esp32h2:23]: @chshu  Can you please verify this. IMO, we can free the BLE memory after commissioning for H2 too in the default cases

**Review by andy31415 — APPROVED** [review]

File: examples/lighting-app/esp32/sdkconfig.defaults.esp32h2 (hunk 1 above)
Comment (chshu) [reply examples/lighting-app/esp32/sdkconfig.defaults.esp32h2:23]: Please keep it for now (with this MR), we will have to double check the BLE deinit feature on H2.

Comment (dhrishi) [reply examples/lighting-app/esp32/sdkconfig.defaults.esp32h2:23]: Talked with @chshu. He mentioned to keep this as is for now. Once we confirm about H2 then we can update it

**Pull request merged**